### PR TITLE
Senlin: Cluster List

### PR DESCRIPTION
**Pull request by JackKuei**
Prior to a PR being reviewed, there needs to be a Github issue that the PR
addresses. Replace the brackets and text below with that issue number.

For #[823 [Add support for Senlin clustering]](https://github.com/gophercloud/gophercloud/issues/823)

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

[[clusters:list]](https://github.com/openstack/senlin/blob/e5bc16b3fea2f10dead365f983c5a937f23d72b0/senlin/api/openstack/v1/clusters.py#L81)

### CODE REVIEW

**Comment by JackKuei** [comment]
recheck

File: openstack/clustering/v1/clusters/requests.go [hunk 1]
```diff
@@ -46,3 +47,40 @@ func Create(client *gophercloud.ServiceClient, opts CreateOptsBuilder) (r Create
...
+type ListOptsBuilder interface {
+	ToClusterListQuery() (string, error)
+}
+
+// ListOpts params
+type ListOpts struct {
+	Limit         int    `q:"limit"`
+	Marker        string `q:"marker"`
+	Sort          string `q:"sort"`
+	GlobalProject string `q:"global_project"`
```
Comment (jtopjian) [inline openstack/clustering/v1/clusters/requests.go]: This looks like a bool value?

https://github.com/openstack/senlin/blob/e5bc16b3fea2f10dead365f983c5a937f23d72b0/senlin/api/openstack/v1/clusters.py#L58-L61

As well, if `false` is a valid value to pass, it should be `*bool`.

**Comment by jtopjian** [comment]
@JackKuei Is this ready to be merged?

**Comment by JackKuei** [comment]
This is ready to be merged

**Comment by JackKuei** [comment]
@jtopjian Ready to be merged.

**Comment by jtopjian** [comment]
@JackKuei Sorry for the delay.

**Pull request merged**